### DISCUSSION TITLE
How to overwrite internal statamic routes?

### DISCUSSION THREAD

**Discussion by torian257x** [statamic/cms#4601] [Q&A] [answered]
I want to overwrite statamic "internal" routes. At the moment auth routes for activation emails etc.
I tried doing it in my web.php but it seems statamic registers its routes before laravel hits my web.php

is there another way to do it?

**Comment by jasonvarga** [comment, marked as answer]
In a service provider you can use the `Statamic::pushWebRoutes` closure. This will add web routes that get loaded before we create our catch-all route.

```php
\Statamic\Statamic::pushWebRoutes(function () {
    Route::get(...);
});
```

**Reply by torian257x** [reply]
i want to reference this https://github.com/statamic/cms/issues/1287#issuecomment-580886853 as well. In case I have any trouble using it correctly
